app/workers/github_repos/repo_sync_worker.rb

```diff
@@ -28,6 +28,11 @@ def perform(repo_id)
         )
         repo.user&.touch(:github_repos_updated_at)
       rescue Github::Errors::NotFound, Github::Errors::Unauthorized
+        repo.destroy
+      rescue Github::Errors::ClientError => e
+        msg = e.message
+        raise e unless msg.include?("Your account was suspended") || msg.include?("Repository access blocked")
```

Comment:
Great catch! BTW this is because we're not wrapping the two correct exceptions for these two statuses:

`Octokit::RepositoryUnavailable` and `Octokit::AccountSuspended`. See https://github.com/octokit/octokit.rb/blob/a69f5eab0a2fb2763fcda9fadcb0c961df96a373/lib/octokit/error.rb#L265 and https://github.com/octokit/octokit.rb/blob/a69f5eab0a2fb2763fcda9fadcb0c961df96a373/lib/octokit/error.rb#L257

If you add two exceptions called `RepositoryUnavailable` and `AccountSuspended` to https://github.com/forem/forem/blob/master/app/services/github/errors.rb you should be able to catch handle them without inspecting the message

Reply:
Oh yeah for sure that would make life easier. going to merge as is and leave that for a refactor. Maybe a good little issue for Hacktoberfest :)

Reply:
Going to make an issue out of the comment, otherwise we'll forget about it :D